packages/grid/src/vaadin-grid-keyboard-navigation-mixin.js

```diff
@@ -593,7 +598,7 @@ export const KeyboardNavigationMixin = (superClass) =>
           // scrolling. Focus the row for the stored focused item index instead.
           const columnIndex = Array.from(targetRow.children).indexOf(this._itemsFocusable);
           const focusedItemRow = Array.from(this.$.items.children).filter(
```

Comment:
nit: could use `.find` instead.

```suggestion
          const focusedItemRow = Array.from(this.$.items.children).find(
            (row) => !row.hidden && row.index === this._focusedItemIndex
          );
```